src/test/java/persistence/entity/entitymanager/EntityManagerTest.java

```diff
@@ -0,0 +1,137 @@
+import persistence.sql.common.DtoMapper;
+import persistence.sql.ddl.querybuilder.CreateQueryBuilder;
+import persistence.sql.dml.querybuilder.SelectQueryBuilder;
+
+import java.util.List;
+import java.util.Optional;
+
+import static persistence.sql.ddl.common.TestSqlConstant.DROP_TABLE_USERS;
+
+class EntityManagerTest {
+    private static final Logger logger = LoggerFactory.getLogger(EntityManagerTest.class);
+    private static DatabaseServer server;
+    private static JdbcTemplate jdbcTemplate;
+
+    private EntityManager entityManager;
+
+    private PersistenceContext persistenceContext;
+    private EntityPersister entityPersister;
+    private EntityLoader entityLoader;
+
+    @BeforeAll
+    static void setupOnce() {
+        try {
+            server = new H2();
+            server.start();
+            jdbcTemplate = new JdbcTemplate(server.getConnection());
+        } catch (Exception e) {
+            logger.error("Error occurred", e);
+        } finally {
+            logger.info("Application finished");
+        }
+    }
+    @BeforeEach
+    void setUp() {
+        String query = new CreateQueryBuilder(Person.class).getQuery();
+        jdbcTemplate.execute(query);
+
+        persistenceContext = new PersistenceContextImpl();
+        entityPersister = new EntityPersister(jdbcTemplate);
+        entityLoader = new EntityLoader(jdbcTemplate);
+        entityManager = new EntityManagerImpl(persistenceContext, entityLoader, entityPersister);
+    }
+
+    @AfterEach
+    void tearDown() {
+        jdbcTemplate.execute(DROP_TABLE_USERS);
+    }
+
+    @AfterAll
+    static void tearDownOnce() {
+        server.stop();
+    }
+
+    @Test
+    void find_entityCache에서_값을_가져온다() {
+        // given
+        Person person = new Person("김철수", 21, "chulsoo.kim@gmail.com", 11);
+        entityManager.persist(person);
+
+        // when
+        Person actual = entityManager.find(Person.class, 1L).get();
+
+        // then
+        Optional<Person> expected = persistenceContext.getEntity(Person.class, person.getId());
+        Assertions.assertThat(actual).isSameAs(expected.get());
+    }
+
+    @Test
+    void find_entityCache와_entityLoader에_값이_없을경우_OptionalEmpty를_반환한다() {
+        // given&when
+        Optional<Person> actual = entityManager.find(Person.class, 1L);
+
+        // then
+        Assertions.assertThat(actual).isEqualTo(Optional.empty());
+    }
+
+    @Test
+    void find_entityCache에_값이_없고_entity_loader에_존재하면_entityloader를_통해_값을_가져오고_entityCache에도_값을_넣는다() {
+        // given&when
+        entityPersister.insert(new Person("김철수", 21, "chulsoo.kim@gmail.com", 11));
+        Optional<Person> actual = entityManager.find(Person.class, 1L);
+
+        // then
+        SoftAssertions softAssertions = new SoftAssertions();
+        softAssertions.assertThat(actual).isSameAs(entityLoader.find(Person.class, 1L));
+        softAssertions.assertThat(actual).isSameAs(persistenceContext.getEntity(Person.class, 1L));
```

Comment:
softAssertions 를 사용해주셨네요 👍 사용법이 조금 잘못되어 검증이 제대로 안되는것 같아요 😅
사용법 참고 링크입니다! https://zzang9ha.tistory.com/418

```suggestion
        // then
        SoftAssertions.assertSoftly(softly -> {
            softly.assertThat(actual.get()).isSameAs(entityLoader.find(Person.class, 1L).get());
            softly.assertThat(actual.get()).isSameAs(persistenceContext.getEntity(Person.class, 1L).get());
        });
```

위의 코드로 바꿔서 테스트시 실패하고 있어요 😢
`softly.assertThat(actual.get()).isSameAs(entityLoader.find(Person.class, 1L).get());` 이 부분이 실패를 하는데, `entityLoader.find` 는 매번 db 에서 select 하기 때문에 `entityManager.find` 와 매번 다른 인스턴스 일 것 같아요 :)